src/ui/utils/launchdarkly.ts

```diff
@@ -1,28 +1,41 @@
 import * as LDClient from "launchdarkly-js-client-sdk";
+import { useEffect, useState } from "react";
 import { UserInfo } from "ui/hooks/users";
 import { isDevelopment } from "./environment";
 
 let client: LDClient.LDClient;
-let ready = false;
+let resolveReady: (ready: boolean) => void;
+const readyPromise = new Promise<boolean>(resolve => {
+  resolveReady = resolve;
+});
 const LD_KEY = isDevelopment() ? "60ca05fb43d6f10d234bb3ce" : "60ca05fb43d6f10d234bb3cf";
 
 function initLaunchDarkly(user?: UserInfo) {
-  ready = false;
   client = LDClient.initialize(LD_KEY, {
     key: user ? user.id : "anon",
   });
 
   client.on("ready", () => {
-    ready = true;
+    resolveReady(true);
   });
 }
 
-function getFeatureFlag(name: string, defaultValue: any) {
-  if (!ready) {
-    return defaultValue;
+function useLaunchDarkly() {
+  const [ready, setReady] = useState(false);
+
+  useEffect(() => {
+    readyPromise.then(r => setReady(r));
+  }, [setReady]);
+
+  function getFeatureFlag(name: string, defaultValue: any) {
```

Comment:
`name: string` can be narrowed to a more precise union of string literals.

It could also be combined with a predefined set of default values.

```ts
const DEFAULT_VALUES = {
  "maintenance-mode": false
} as const;

function getFeatureFlag<F extends keyof DEFAULT_VALUES>(flagName: F) {
   if (!ready) {
     return DEFAULT_VALUES[flagName]
   }
   return client.variation(name, DEFAULT_VALUES[flagName])
}
```